### DISCUSSION TITLE
[Help Wanted] Generating keys forever and asks for keychain authentication (restore keychain)

### DISCUSSION THREAD

**Discussion by milkywaymetropolis** [standardnotes/forum#2804] [General]
**Describe the issue**
I cannot login into standard notes app on android. It says **"generating keys"** forever while logging in. Like spent 1 hour on generating keys with no output.

Tried clearing app cache, app data and reinstalled. Tried all these three steps individually as well as together. Sadly with no result. The issue is still there.

When I closed the app and reopened without clearing anything, it asks for **"authentication for keychain"**. Same happens when I just clear cache only. I gave correct password every time. I double checked.. there is no error on password.

Tried a few tips from github issue page as well... with no result again.

However the web app and the desktop apps work fine without any issues. And when I check the login sessions from other devices, it shows Android 5.0 is logged in but this error is persistent. Cannot access a single note on that device.

I changed password recently and after that this error is persistent no matter what on my Android 5.0 device.

Device OS: Android 5.0 Lollipop

Please help me with this issue. This is quite annoying.

Best regards.

**To Reproduce**
1. Go to 'Settings'
2. Click on 'Sign in'
3. Give 2FA
4. See error (generating keys forever + keychain error)

**Expected behavior**
Generate keys within seconds and go to usual notes interface.

**Smartphone (please complete the following information):**
 - Device: Samsung J2
 - OS: Android 5.0 Lollipop
 - Browser: Stock browser

**Additional context**
This issue never happened before until I changed the account password recently.

**Comment by ghost** [comment]
I have the same problem, albeit intermittently, on both Android 13 and Windows 10.  The app will say my keychain has been wiped (even though no action has been taken that should cause this) and upon entering my password the app sits for hours "Generating keys".  Eventually after several app/device restarts I will be logged in when opening the app.

**Comment by effieeee** [comment]
Hi @synthc, that's odd if you haven't done anything in particular... may I know if you've tried fully resetting both apps by following the corresponding instructions for each platform in [this link](https://docs.standardnotes.com/troubleshooting/reset-apps/)?

**Comment by ghost** [comment]
> Hi @synthc, that's odd if you haven't done anything in particular... may I know if you've tried fully resetting both apps by following the corresponding instructions for each platform in [this link](https://docs.standardnotes.com/troubleshooting/reset-apps/)?

Yes, I have.

**Comment by effieeee** [comment]
I see, can I also ask which versions of the app you're running on your Android and Windows devices? Please also let me know your account email by sending us a message at help@standardnotes.com 🙏

**Comment by ghost** [comment]
Windows version:  3.100.17
Android:  3.46.22

**Comment by effieeee** [comment]
Thank you for providing the info requested! I noticed your account is also fairly recent, so I'm sorry that you're encountering this sort of issue right away :( Can I now ask you to try the following next:

1. Sign out of your app on Android.
2. On your Windows app, go to **Preferences > Account**, then choose **Sign out other sessions**
![image](https://user-images.githubusercontent.com/54989748/204034992-1a891cb9-f6b5-42b9-9a73-7d1b265b87f7.png)
3. Sign out of the Windows app as well, then sign back in again on this device and your Android device as well.
4. Monitor the situation again and let me know if you still encounter the reauthentication prompts

**Comment by ghost** [comment]
I followed those steps but I'm still getting prompts (on Windows at least, haven't seen it reoccur on Android yet) and they still get stuck generating keys indefinitely.

**Comment by ghost** [comment]
This is still happening very regularly (around 50% of time), making the Windows app nearly unusable.
